extensions/amp-story-panning-media/0.1/amp-story-panning-media.js

```diff
@@ -113,20 +129,50 @@ export class AmpStoryPanningMedia extends AMP.BaseElement {
       // TODO(#31932): A key could be sent here to update elements of the same group.
       // Note, this will not work when there are 2 or more panning components on the same page.
       // It might need to dynamic to hold more than 1 set of positions.
-      this.storeService_.dispatch(Action.SET_PANNING_MEDIA_STATE, {
-        x: this.x_,
-        y: this.y_,
-        zoom: this.zoom_,
-      });
+      this.animate_();
     }
   }
 
+  /**
+   * Animates from current position to active components position.
+   * @private
+   */
+  animate_() {
+    const startPos = this.storeService_.get(
+      StateProperty.PANNING_MEDIA_STATE
+    ) || {x: this.x_, y: this.y_, zoom: this.zoom_};
+
+    const easeInOutQuad = (val) =>
+      val < 0.5 ? 2 * val * val : 1 - Math.pow(-2 * val + 2, 2) / 2;
+
+    let frame = 0;
+    const nextFrame = () => {
+      frame++;
+      if (frame < MAX_FRAMES) {
```

Comment:
Can you explain a bit more this implementation?

If I understand correctly, the animation will last for 1 second, so you'll call rAF and assume that we do get 60 frames per second.
I think this assumption is very risky, as devices might drop frames, and we don't know if all devices actually run at 60fps (remember the Pixel phone 90 Hz screen)

Reply:
Thank you for pointing this out. I refactored this to use the timestamp returned from the rAF callback.
This now follows the pattern suggested in the [MDN](https://developer.mozilla.org/en-US/docs/Web/API/window/requestAnimationFrame) docs.

This could be updated in the 360 component as a follow up.